drivers/watchdog/wdt_nxp_ewm.c

```diff
@@ -0,0 +1,199 @@
+/*
+ *
+ * Copyright 2025 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT nxp_ewm
+
+#include <zephyr/drivers/watchdog.h>
+#include <zephyr/irq.h>
+#include <fsl_clock.h>
+
+#define LOG_LEVEL CONFIG_WDT_LOG_LEVEL
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(wdt_nxp_ewm);
+
+#define NXP_EWM_FEED_MAGIC_NUMBER 0x2CB4
+#define NXP_EWM_MAX_TIMEOUT_WINDOW 0xFE
+
+struct nxp_ewm_config {
+	EWM_Type *base;
+	void (*irq_config_func)(const struct device *dev);
+	bool is_input_enabled;
+	bool is_input_active_high;
+	uint8_t clk_divider;
+};
+
+struct nxp_ewm_data {
+	struct wdt_timeout_cfg timeout_cfg;
+	bool is_watchdog_setup;
+};
+
+static int nxp_ewm_setup(const struct device *dev, uint8_t options)
+{
+	const struct nxp_ewm_config *config = dev->config;
+	struct nxp_ewm_data *data = dev->data;
+	EWM_Type *base = config->base;
+
+	if (!data->is_watchdog_setup) {
```

Comment:
similar comment, invert the condition and return -EBUSY if needed, then put the rest of the code. Minimize nesting

Reply:
updated